### PR TITLE
[CIS-1818, CIS-1806] Do not remove pending messages from channel when updating it

### PR DESCRIPTION
**Pull request by polqf**
### 🔗 Issue Links
https://stream-io.atlassian.net/browse/CIS-1806
https://stream-io.atlassian.net/browse/CIS-1818

### 🎯 Goal

Make sure we don't lose messages with attachments when synchronize is called.

### 📝 Summary

The order of execution when sending a message with attachments is the following:
1. Attachment is sent
2. Message is sent together with the attachment information

Given the above, step 2 is delayed based on how big the attachment is. 

When calling `synchronize` on ChannelController, it was removing all the message relationships with the channel. That meant all messages were still in DB, but without a channel. That led to a case where all the messages that were in .pendingSend waiting for the attachments to be uploaded were removed from the channel in DB. When the attachments were uploaded, and MessageSender was trying to send the message, because of the missing Channel relationship, it was unable to place a request given that this is a required parameter

### 🛠 Implementation

Now, ChannelController is only removing the relationships with messages that are not pending. That way, whenever the attachment uploads are finished, we can properly execute the requests to send the message.

### 🧪 Manual Testing Notes

1. Upload a big file
2. Quickly go back to the channel list
3. Quickly open the channel again

Expected result:
The message containing the attachment should still be visible, and the upload should complete as normal

### ☑️ Contributor Checklist
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] This change follows zero ⚠️ policy (required)
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChat/Workers/ChannelUpdater.swift [hunk 1]
```diff
@@ -32,9 +32,8 @@ class ChannelUpdater: Worker {
                 let payload = try result.get()
                 channelCreatedCallback?(payload.channel.cid)
                 database?.write { session in
-                    if clearMessageHistory {
-                        let channelDTO = session.channel(cid: payload.channel.cid)
-                        channelDTO?.messages.removeAll()
+                    if clearMessageHistory, let channelDTO = session.channel(cid: payload.channel.cid) {
+                        channelDTO.messages = channelDTO.messages.filter { $0.localMessageState?.isPending == true }
```
[the remark this answers is not in the document]
Comment (polqf) [reply Sources/StreamChat/Workers/ChannelUpdater.swift:36]: I don't think those should be kept. WDYT?

[the remark this answers is not in the document]
Comment (evsaev) [reply Sources/StreamChat/Workers/ChannelUpdater.swift:36]: If I found a nice giphy but had to answer in another channel, I would get annoyed by losing it.  Let's check how other SDKs handle this?

[the remark this answers is not in the document]
Comment (polqf) [reply Sources/StreamChat/Workers/ChannelUpdater.swift]: isn't this a super edge case? Also, what's the state of the message in this case? I bet it fits the requirements to be kept based on the current logic

[the remark this answers is not in the document]
Comment (evsaev) [reply Sources/StreamChat/Workers/ChannelUpdater.swift]: Yeah, but why not to cover it? 🙂 The local state becomes `nil` right after sending the message

[the remark this answers is not in the document]
Comment (polqf) [reply Sources/StreamChat/Workers/ChannelUpdater.swift]: If there is an issue with Giphy messages, I don't know tbh. But we should leave that to another PR/ticket

**Review by evsaev — APPROVED** [review]

File: Sources/StreamChat/Workers/ChannelUpdater.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (evsaev) [reply Sources/StreamChat/Workers/ChannelUpdater.swift]: Fixing gifs is a one-liner that fits the scope. But if that's an issue, let's merge and have a separate task

**Review by bielikb — APPROVED** [review]

**Pull request merged**